### PR TITLE
feat(build): add --extract-css flag

### PR DESCRIPTION
**Pull request by filipesilva**
Having css extraction on all the time makes serve rebuild times very slow for global styles.

This flag defaults to false on dev targets and true on prod targets for `ng serve`, and defaults to false in `ng build`.

@intellix can you verify this solves your long reloads for scss?

Partially address https://github.com/angular/angular-cli/issues/1980

### CODE REVIEW

**Comment by intellix** [comment]
Great improvement :)

b24:
 **styles.scss**: ~12s
 **my-cmp.scss**: ~3.5 - 5s

this PR:
 **styles**: ~5-6s
 **some-cmp**: ~3s

File: packages/angular-cli/commands/serve.run.ts [hunk 1]
```diff
@@ -20,6 +20,9 @@ export default function serveRun(commandOptions: ServeTaskOptions) {
     }
   }
 
+  // default to extractCss to true on prod target
+  commandOptions.extractCss = commandOptions.extractCss || commandOptions.target === 'production';
```
Comment (clydin) [inline packages/angular-cli/commands/serve.run.ts]: Won't this always be true for production?

**Comment by montella1507** [comment]
@filipesilva hello, does it mean, that any change in global CSS file (or scss file) WILL NOT proc entire rebuild process? It will be just recompiled by something like node-sass and the page will be refreshed? (would it be possible to hot-reload/live reload styles?)

File: packages/angular-cli/commands/serve.run.ts (hunk 1 above)
Comment (filipesilva) [inline packages/angular-cli/commands/serve.run.ts]: Right, bad logic there. Will fix.

**Comment by filipesilva** [comment]
@montella1507 for global CSS, yes. Component CSS still will need reloads.

**Comment by montella1507** [comment]
@filipesilva  oh jesus, this would be the best commit ever :D if it works, it would be absolutely perfect :-) My HTML/CSS designer will love you for that, because he wont need to rebuild entire page with every little change :-)

**Comment by hansl** [comment]
LGTM.

**Comment by Meligy** [comment]
Does this mean no global CSS watch in dev?

**Comment by filipesilva** [comment]
@Meligy I don't quite understand your question. Can you rephrase?

**Comment by Meligy** [comment]
Does this PR affect the default behaviour of watching global CSS files and rebuilding (re-bundling) them on change when running `ng build` / `ng serve` in dev target/env?

**Comment by filipesilva** [comment]
@Meligy nope. But the css will be served in js bundles instead of css files in `ng serve`, which was how it worked since beta 15 or something. If you absolutely need the css files in dev rebuilds, do `ng serve --extract-css`. It's gonna be slower though.

**Comment by Meligy** [comment]
Aha. Makes sense. Good stuff!

**Comment by montella1507** [comment]
i have done some benchmarks:

gulp-sass  (build and copy generated .css to src folder)+ angular-cli (watch detech + rebuild):
150ms + 18sec


angular-cli only (styles:["style.scss"]) + --extract-css:
35sec 
¨:-/

**Comment by filipesilva** [comment]
@montella1507 extracting CSS in webpack is always slow I believe :/
